### DISCUSSION TITLE
How to turn a BarStack into a BarStackHorizontal

### DISCUSSION THREAD

**Discussion by jimmy-e** [airbnb/visx#1207] [Q&A]
How do I take a `BarStack` and rotate it so that it is horizontal? I am currently doing something like the following, but it does not appear to be working:

```js
import React from 'react';
import { BarStack } from '@visx/shape';

export default ({ accessor, xScale, yScale }) => (
  <BarStack<Datum, string>
    x={accessor}
    xScale={xScale}
    yScale={yScale}
  >
    ...
  </BarStack>
);
```

```js
import React from 'react';
import { BarStackHorizontal } from '@visx/shape';

export default ({ accessor, xScale, yScale }) => (
  <BarStackHorizontal<Datum, string>
    y={accessor}
    xScale={yScale}
    yScale={xScale}
  >
    ...
  </BarStackHorizontal>
);
```

One thing that makes this less clear to find out on my own is that the gallery example for the [Bar Stack (vertical)](https://airbnb.io/visx/barstack) and the [Bar Stack Horizontal](https://airbnb.io/visx/barstackhorizontal), when analyzed in a diffchecker, are not too similiar, making it confusing as to what is the important difference between these two examples. Thanks for the help!

**Comment by VigneshTrantor** [comment]
@jimmy-e 
I was able to do it by flipping the XY chart xScale and yScale values "{ type: "band" }" and "{ type: "linear" }". Just interchange them.

Also, you will need to flip the accessors
export const DATUM_ACCESSORS = (isVertical: boolean) => ({ xAccessor: (d: any) => (isVertical ? d.x : d.y), yAccessor: (d: any) => (isVertical ? d.y : d.x), });

We had a requirement to be able to flip the chart on clicking of the radio button.

Here is a codesandbox example https://codesandbox.io/s/barcharts-horizontal-v09d2?file=/Example.tsx

**Reply by jimmy-e** [reply]
Thanks @VigneshTrantor ! This seems though to apply to the XY Chart, do you know if this still applies to the `BarStack` and `BarStackHorizontal` api's for `@visx/shape`?

**Reply by VigneshTrantor** [reply]
I don't know for sure. Better to make separate components for horizontal and vertical charts and render them conditionally.

**Comment by maximerivolet** [comment]
Hi @jimmy-e, I already asked this here : https://github.com/airbnb/visx/discussions/1050

I hope it will help you !

**Reply by jimmy-e** [reply]
Thanks! See my other response above.
